File: content-solutions/two-pointers/palindrome-note.java
class Solution {
    public boolean solution(String note) {
        int l = 0, r = note.length() - 1;
        while (l < r) {
            char a = note.charAt(l), b = note.charAt(r);
            if (!Character.isLetterOrDigit(a)) l++;
            else if (!Character.isLetterOrDigit(b)) r--;
            else if (Character.toLowerCase(a) != Character.toLowerCase(b)) return false;
            else {
                l++;
                r--;
            }
        }
        return true;
    }
}

File: content-solutions/two-pointers/palindrome-note.js
function solution(note) {
  const ok = (c) => /[a-z0-9]/i.test(c);
  let l = 0;
  let r = note.length - 1;
  while (l < r) {
    if (!ok(note[l])) l++;
    else if (!ok(note[r])) r--;
    else if (note[l].toLowerCase() !== note[r].toLowerCase()) return false;
    else {
      l++;
      r--;
    }
  }
  return true;
}
